src/hooks/useConvert.jsx

```diff
@@ -1,72 +1,78 @@
 import React from 'react';
+import { parseCSV, stringifyCSV, parseJson, generateFileName, fetchGeoData, insertGeoData } from '../utils/csv';
 
 const useConvert = () => {
   const [file, setFile] = React.useState(undefined);
-  const [isConvertToJson, setIsConvertToJson] = React.useState(false);
-  const [jsonData, setJsonData] = React.useState([]);
+  const [csvText, setCsvText] = React.useState('');
+  const [jsonText, setJsonText] = React.useState('');
 
   const fileReader = new FileReader();
 
   const handleUploadFileChange = (event) => {
     setFile(event.target.files[0]);
-    setIsConvertToJson(false);
-  };
-
-  const convertCsvFileToJson = React.useCallback((text) => {
-    const csvHeader = text.slice(0, text.indexOf('\n')).replace('\r', '').split(',');
-    let csvData = text
-      .slice(text.indexOf('\n') + 1)
-      .split('\n')
-      .map((s) => s.replace('\r', '').split(','));
+    fileReader.onload = function (event) {
+      const text = event.target.result;
+      setCsvText(text);
+    };
 
-    csvData.splice(-1);
-
-    const convertData = csvData.map((data) => {
-      let newObject = {};
-      csvHeader.forEach((header, index) => {
-        newObject = { ...newObject, [header]: data[index] };
-      });
-      return newObject;
-    });
+    fileReader.readAsText(event.target.files[0]);
+  };
 
-    setJsonData(convertData);
+  const convertCsvFileToJson = React.useCallback((csvText) => {
+    const { csvHeader, csvData } = parseCSV(csvText);
+    const json = stringifyCSV({ csvHeader, csvData });
+    setJsonText(json);
   }, []);
 
   const handleCsvToJson = () => {
-    if (file) {
-      fileReader.onload = function (event) {
-        const text = event.target.result;
-        convertCsvFileToJson(text);
-      };
-
-      fileReader.readAsText(file);
-      setIsConvertToJson(true);
-    }
+    convertCsvFileToJson(csvText);
   };
 
   const convertJsonToCsvFile = () => {
-    const headers = Object.keys(jsonData[0]);
+    const text = parseJson(jsonText);
+    setCsvText(text);
```

Comment:
implement `stringifyCSV` with the same signature as `JSON.stringify`, then this becomes: 

```suggestion
    const data = JSON.parse(jsonText);
    setCsvText(stringifyCSV(data));
```